src/main/java/org/eclipse/cargotracker/domain/model/cargo/Itinerary.java

```diff
@@ -23,10 +19,9 @@
   private static final long serialVersionUID = 1L;
 
   // TODO [Clean Code] Look into why cascade delete doesn't work.
-  @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+  @OneToMany(cascade = CascadeType.ALL)
   @JoinColumn(name = "cargo_id")
-  // TODO [Clean Code] Index this is in leg_index
-  @OrderBy("loadTime")
+  @OrderColumn(name = "leg_index")
```

Comment:
Could you please explain why this change was important? Is this an issue with Hibernate?

Reply:
`getLegs` does not apply same orders as before inserting in Hibernate.

Reply:
`getLegs` does not apply same order as before inserting in Hibernate.

Reply:
`getLegs` does not apply same order as before inserting in Hibernate.

Reply:
https://stackoverflow.com/questions/11433195/hibernate-orderby-vs-ordercolumn-to-maintain-the-order-of-a-collection